include/utils/export.php

```diff
@@ -97,115 +97,58 @@ function export($type, $format = 'CSV') {
-		}
+	if (isset($_SESSION['list_query'])) {
+		$query = $_SESSION['list_query'];
 	}
-	$params = array();
-
-	if (($search_type == 'withoutsearch' || $search_type == 'includesearch') && $export_data == 'selecteddata') {
-		$idstring = explode(';', vtlib_purify($_REQUEST['idstring']));
-		if ($type == 'Accounts' && count($idstring) > 0) {
-			$query .= ' and vtiger_account.accountid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'Contacts' && count($idstring) > 0) {
-			$query .= ' and vtiger_contactdetails.contactid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'Potentials' && count($idstring) > 0) {
-			$query .= ' and vtiger_potential.potentialid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'Leads' && count($idstring) > 0) {
-			$query .= ' and vtiger_leaddetails.leadid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'Products' && count($idstring) > 0) {
-			$query .= ' and vtiger_products.productid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'Documents' && count($idstring) > 0) {
-			$query .= ' and vtiger_notes.notesid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'HelpDesk' && count($idstring) > 0) {
-			$query .= ' and vtiger_troubletickets.ticketid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'Vendors' && count($idstring) > 0) {
-			$query .= ' and vtiger_vendor.vendorid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'Invoice' && count($idstring) > 0) {
-			$query .= ' and vtiger_invoice.invoiceid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'Quotes' && count($idstring) > 0) {
-			$query .= ' and vtiger_quotes.quoteid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'SalesOrder' && count($idstring) > 0) {
-			$query .= ' and vtiger_salesorder.salesorderid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif ($type == 'PurchaseOrder' && count($idstring) > 0) {
-			$query .= ' and vtiger_purchaseorder.purchaseorderid in ('. generateQuestionMarks($idstring) .')';
-			$params[] = $idstring;
-		} elseif (count($idstring) > 0) {
-			// vtlib customization: Hook to make the export feature available for custom modules.
-			$query .= " and $focus->table_name.$focus->table_index in (" . generateQuestionMarks($idstring) . ')';
-			$params[] = $idstring;
-			// END
+	if ($export_data == 'currentpage') {
+		$query .= $_SESSION['limitQuery'];
+	} elseif ($export_data == 'selecteddata') {
+		$idsArray = explode(';', vtlib_purify($_REQUEST['idstring']));
+		$where = '';
+		foreach ($idsArray as $key => $value) {
+			if (!empty($value)) {
```

Comment:
why isnt this an `in ('. generateQuestionMarks($idstring) .')` ?  better as a list of ORs?